homeassistant/config_entries.py

```diff
@@ -803,13 +803,11 @@ async def async_unload(
             assert isinstance(result, bool)
 
             # Only adjust state if we unloaded the component
-            if domain_is_integration and result:
-                await self._async_process_on_unload(hass)
-                if hasattr(self, "runtime_data"):
-                    object.__delattr__(self, "runtime_data")
-
-                self._async_set_state(hass, ConfigEntryState.NOT_LOADED, None)
+            if domain_is_integration:
```

Comment:
Please don't change the core in this PR.

Reply:
Oops, forgot extra commits were added to my other PR, #119224. I've reverted